### ISSUE TITLE
[MD007] `start-indented` does not play well with mixed lists (text-aligned)

### ISSUE THREAD

**Issue by robsdedude** [rvben/rumdl#541]
### rumdl version

0.1.58

### Installation method

Pre-built binary

### Bug description

When setting `start-indented = true`, mixed lists (OL + UL) don't respect text-aligned styling.

### Steps to reproduce

1. Create `.rumdl.toml`
    ```toml
    [MD007]
    start-indented = true
    # style = "text-aligned"  # uncommenting this makes no difference
    # start-indent = 2  # default value, uncommenting is equivalent
    ```
2. Create `test.md`
    ```markdown
    # Le Title
    
      1. First thing
         * Sub thing
    ```
3. Run `rumdl` and see
    ```text
    test.md:4:1: [MD007] Expected 4 spaces for indent depth 1, found 5 [*]
    
    Issues: Found 1 issues in 1 file (13ms)
    ```

### Expected behavior

Don't complain about `test.md`

Especially problematic is that `rumdl` suggests to "fix" the file to

```markdown
# Le Title

  1. First thing
    * Sub thing
```

which isn't properly rendered (at least by GH's markdown engine).

### Actual behavior

*see reproducer step 3*

### Operating system

Linux

### Additional Observation

<details>

<summary><code>text-aligned</code> works as expected without <code>start-indented</code></summary>

Configuring `start-indented = false` (or commenting it out, as it's the default) and de-indenting the whole list by 2 spaces makes the lint go away.

`.rumdl.toml`
``` toml
[MD007]
start-indented = false
```

```markdown
# Le Title

1. First thing
   * Sub thing
```

Yields no lints. So `text-aligned` styling seems to be working as expected in this case.

</details>

**Comment by robsdedude** [comment]
@rvben Do you happen to already be actively working on this? If not, I'd like to have a look to see if I can cook up a PR (no promises that I'll ultimately manage to).

**Comment by rvben** [comment]
Thanks for the offer, @robsdedude! Yesterday I already worked on this. It'll be fixed in the upcoming release.

**Comment by rvben** [comment]
Fixed in v0.1.60. The `start_indented` option now correctly respects text-aligned indentation for unordered lists nested under ordered lists.

**Comment by robsdedude** [comment]
Thanks a whole bunch for your consistent and surprisingly fast fixes. And generally for this helpful piece of software.

**Comment by rvben** [comment]
Thanks! And I am very open to PRs, so if you see something that can be improved feel free to raise one 🙂
